src/Symfony/Component/Cache/Simple/Psr6Cache.php

```diff
@@ -11,218 +11,13 @@
-    }
-
-    /**
-     * {@inheritdoc}
-     */
-    public function get($key, $default = null)
-    {
-        try {
-            $item = $this->pool->getItem($key);
-        } catch (SimpleCacheException $e) {
-            throw $e;
-        } catch (Psr6CacheException $e) {
-            throw new InvalidArgumentException($e->getMessage(), $e->getCode(), $e);
-        }
-
-        return $item->isHit() ? $item->get() : $default;
-    }
-
-    /**
-     * {@inheritdoc}
-     */
-    public function set($key, $value, $ttl = null)
-    {
-        try {
-            if (null !== $f = $this->createCacheItem) {
-                $item = $f($key, $value);
-            } else {
-                $item = $this->pool->getItem($key)->set($value);
-            }
-        } catch (SimpleCacheException $e) {
-            throw $e;
-        } catch (Psr6CacheException $e) {
-            throw new InvalidArgumentException($e->getMessage(), $e->getCode(), $e);
-        }
-        if (null !== $ttl) {
-            $item->expiresAfter($ttl);
-        }
-
-        return $this->pool->save($item);
-    }
-
-    /**
-     * {@inheritdoc}
-     */
-    public function delete($key)
-    {
-        try {
-            return $this->pool->deleteItem($key);
-        } catch (SimpleCacheException $e) {
-            throw $e;
-        } catch (Psr6CacheException $e) {
-            throw new InvalidArgumentException($e->getMessage(), $e->getCode(), $e);
-        }
-    }
-
-    /**
-     * {@inheritdoc}
-     */
-    public function clear()
-    {
-        return $this->pool->clear();
-    }
-
-    /**
-     * {@inheritdoc}
-     */
-    public function getMultiple($keys, $default = null)
-    {
-        if ($keys instanceof \Traversable) {
-            $keys = iterator_to_array($keys, false);
-        } elseif (!\is_array($keys)) {
-            throw new InvalidArgumentException(sprintf('Cache keys must be array or Traversable, "%s" given', \is_object($keys) ? \get_class($keys) : \gettype($keys)));
-        }
-
-        try {
-            $items = $this->pool->getItems($keys);
-        } catch (SimpleCacheException $e) {
-            throw $e;
-        } catch (Psr6CacheException $e) {
-            throw new InvalidArgumentException($e->getMessage(), $e->getCode(), $e);
-        }
-        $values = array();
-
-        foreach ($items as $key => $item) {
-            $values[$key] = $item->isHit() ? $item->get() : $default;
-        }
-
-        return $values;
-    }
-
-    /**
-     * {@inheritdoc}
-     */
-    public function setMultiple($values, $ttl = null)
-    {
-        $valuesIsArray = \is_array($values);
-        if (!$valuesIsArray && !$values instanceof \Traversable) {
-            throw new InvalidArgumentException(sprintf('Cache values must be array or Traversable, "%s" given', \is_object($values) ? \get_class($values) : \gettype($values)));
-        }
-        $items = array();
-
-        try {
-            if (null !== $f = $this->createCacheItem) {
```

Comment:
The problem I have experienced is that in dev mode the cache is decorated with a TraceableCache. This means it loses this optimization and introduces https://github.com/symfony/symfony/issues/28918#issuecomment-432399481 again

Reply:
Good catch, that should be fixed as a bug!

Reply:
fixed in https://github.com/symfony/symfony/pull/29243